src/window.ts

```diff
@@ -116,19 +119,22 @@ export type SEventsInfos = {
     nightlies: Array<number>;
     night: boolean;
   };
-} & {
-  // Christmas
-  holidayseason?: boolean;
-  grinch?: SMonsterEvent;
-  snowman?: SMonsterEvent;
-} & {
-  crabxx?: SMonsterEvent;
-};
+} & Partial<Record<EventKey, SMonsterEvent>> & {
+    // Christmas
+    holidayseason?: boolean; // unsure if this will override the above general purpose EventKey
```

Comment:
I think it will try to create an intersection and become `boolean & SMonsterEvent` which will be `never` and the property will be broken.

Reply:
I'll have to test the outcome then, we don't want players to think holidayseason is anything else than a bool